modules/heat_conduction/src/kernels/ADMatHeatSource.C

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADMatHeatSource.h"
+
+registerADMooseObject("HeatConductionApp", ADMatHeatSource);
+
+defineADValidParams(
+    ADMatHeatSource,
+    ADKernel,
+    params.addParam<Real>("scalar", 1.0, "Scalar multiplied by the body force term");
+    params.addRequiredParam<MaterialPropertyName>("material",
```

Comment:
I would rename this to whatever you rename the kernel member (minus the leading underscore)